app/src/main/java/com/kickstarter/viewmodels/CommentsViewModel.kt

```diff
@@ -181,6 +183,31 @@ interface CommentsViewModel {
 
             loadCommentListFromProjectOrUpdate(projectOrUpdateComment)
 
+            val deepLinkCommentableId =
+                intent().filter {
+                    it.getStringExtra(IntentKey.COMMENT)?.isNotEmpty()
+                }.map { requireNotNull(it.getStringExtra(IntentKey.COMMENT)) }
+
+            deepLinkCommentableId
+                .compose(Transformers.takeWhen(projectOrUpdateComment))
+                .switchMap {
+                    return@switchMap apolloClient.getComment(it)
+                }.compose(Transformers.neverError())
+                .compose(combineLatestPair(deepLinkCommentableId))
+                .map {
+                    CommentCardData.builder()
+                        .comment(it.first)
+                        .project(this.project)
+                        .commentCardState(it.first.cardStatus())
+                        .commentableId(it.second)
+                        .build()
+                }.compose(bindToLifecycle())
+                .subscribe {
```

Comment:
nit: i think we can condense this into one line: 
`.subscribe(this.startThreadActivityFromDeepLink)`